src/Common/Exception.h

```diff
@@ -40,7 +40,11 @@ class Exception : public Poco::Exception
     const char * what() const throw() override { return message().data(); }
 
     /// Add something to the existing message.
-    void addMessage(const std::string & arg) { extendedMessage(arg); }
+    template <typename ...Fmt>
+    void addMessage(Fmt&&... fmt)
+    {
+        extendedMessage(fmt::format(std::forward<Fmt>(fmt)...));
+    }
```

Comment:
That's not a good interface, I suppose.
It is not clear that we can't call it with zero arguments. Format function is declared like `format(const S& format_str, Args&&... args)` .
Also, it have changed prev behaviour when we add single string. That's why we have `invalid format string` message in unit test. Now, it is not just a string, it must be `format` string.

I think it is better to add separate method `addFormattedMessage(const S& format_str, Args&&... args)` or `addFmtMessage(const char *, Args && ... args)` and keep previous one. (We can also overload `addMessage` with templates, but it would be less obvious).

Reply:
You're right, added a separate function. Let's use the overload because it's more convenient -- when you want to edit the message and add some parameters, no need to rename the function. Also consistent with `LOG_***` macros.

Also did the same for the Exception constructor -- it had the same problem.